ls8/cpu.c

```diff
@@ -59,8 +63,48 @@ void alu(struct cpu *cpu, enum alu_op op, unsigned char regA, unsigned char regB
 void cpu_run(struct cpu *cpu)
 {
   int running = 1; // True until we get a HLT instruction
-
   while (running) {
+    /* int command = 0; */
+    /* for(int i = 0; i<10; i++) { */
+    /*  printf("ram stuff => %x\n", cpu->ram[i] ) ; */
+    /* } */
+    unsigned int command = cpu->ram[cpu->pc];
+    /* unsigned int command */
+    /* printf("command => %d\n", command); */
+    switch(command) {
+/* printf(" inside switch command\n"); */
+      case HLT:
+        /* printf("HLT command\n"); */
+        running = 0;
+        break;
+
+      case HELLO:
+        printf("hello world\n");
+        /* cpu->pc++; */
+        break;
+
+      case LDI:
+        /* printf("ldi is working\n"); */
+        /* set the value of a register to an integer */
+       /* printf("BEFORE cpu -> pc %d\n", cpu->pc ) ; */
+       /* printf("BEFORE cpu -> ram %d\n", cpu->ram[cpu->pc]) ; */
+        cpu->registers[cpu->ram[cpu->pc + 1]] = cpu->ram[cpu->pc + 2];
```

Comment:
if I may suggest: you can store these into variables to make them easier to read when you're coming back to these. so, something like
`arg_1 = cpu->ram[cpu->pc + 1]`
`arg_2 = cpu->ram[cpu->pc + 2]`
`cpu->registers[arg_1] = arg_2`

it's a few extra lines, but it'll make it muuuch easier to debug